luma_core/src/ipc.rs

```diff
@@ -1,26 +1,26 @@
-use core::ptr::from_exposed_addr_mut;
+use core::ptr;
 
 //bits 0..=31 = physical address of ipc request
-const HW_IPC_PPCMSG: usize = 0xCD00_0000usize; //from_exposed_addr_mut(0xCD00_0000);
+const HW_IPC_PPCMSG: *mut u32 = ptr::without_provenance_mut(0xCD00_0000);
 
 //bit 0 = X1 | Execute IPC request
 //bit 1 = Y2 | Acknowledge IPC request
 //bit 2 = Y1 | IPC request reply available
 //bit 3 = X2 | Relaunch IPC
 //bit 4 = IY1 | IPC request reply send out IPC interrupt
 //bit 5 = IY2 | IPC request acknowledge sends out IPC interrupt
-const HW_IPC_PPCCTRL: usize = 0xCD00_0004usize; //from_exposed_addr_mut(0xCD00_0004);
+const HW_IPC_PPCCTRL: *mut u32 = ptr::without_provenance_mut(0xCD00_0004);
```

Comment:
we want `with_provenance_mut` here instead of `without_provenance_mut` otherwise writing to value is UB

Reply:
`with_provenance_mut()` is also UB, since we haven’t converted any pointer to an address right before, because we don’t have any pointer to convert. In addition, it isn’t `const` so we couldn’t use it here.